### PR TITLE
fix(JOML): align with BlockRegion API changes

### PR DESCRIPTION
**Pull request by jdrueckert**

### CODE REVIEW

File: src/main/java/org/terasology/volcanoes/SurfaceRasterizer.java [hunk 1]
```diff
@@ -40,10 +41,10 @@ public void initialize() {
...
     public void generateChunk(CoreChunk chunk, Region chunkRegion) {
         ElevationFacet elevationFacet = chunkRegion.getFacet(ElevationFacet.class);
-        for (Vector3i position : chunkRegion.getRegion()) {
-            float surfaceHeight = elevationFacet.getWorld(position.x, position.z);
-            if (position.y < surfaceHeight) {
-                chunk.setBlock(ChunkMath.calcRelativeBlockPos(position), stone);
+        for (Vector3ic position : chunkRegion.getRegion()) {
+            float surfaceHeight = elevationFacet.getWorld(position.x(), position.z());
+            if (position.y() < surfaceHeight) {
+                chunk.setBlock(ChunkMath.calcRelativeBlockPos(position, new Vector3i()), stone);
```
Comment (skaldarnar) [inline src/main/java/org/terasology/volcanoes/SurfaceRasterizer.java:47]: consider using a `tmp` vector here for reuse.

File: src/main/java/org/terasology/volcanoes/VolcanoRasterizer.java [hunk 2]
```diff
@@ -16,9 +16,10 @@
 
 package org.terasology.volcanoes;
 
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+import org.lwjgl.system.CallbackI;
```
Comment (skaldarnar) [inline src/main/java/org/terasology/volcanoes/VolcanoRasterizer.java:21]: stray import?

**Review by skaldarnar — APPROVED** [review]

File: src/main/java/org/terasology/volcanoes/SurfaceRasterizer.java (hunk 1 above)
Comment (skaldarnar) [reply src/main/java/org/terasology/volcanoes/SurfaceRasterizer.java:47]: Oh, and I think there's `Chunks.toRelative(position, ...)` as a replacement for `ChunkMath`.

**Review by Cervator — APPROVED** [review]
Tested locally, addressing code review comments along with the merge :-)

Thanks!

**Pull request merged**